Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitClient.swift

```diff
@@ -55,8 +75,21 @@ import BraintreeCore
             if error != nil {
                 completion(nil, error)
                 return
-            } else if result != nil {
-                // TODO: future PR start ASWebAuthenticationSession with result.approvalURL
+            } else if result != nil, let result = result {
+                if result.approvalURL == "null" {
+                    // TODO: call tokenize - url already approved
+                } else if let url = URL(string: result.approvalURL) {
+                    self.startAuthenticationSessionWithoutContext(url: url) { success, error in
+                        switch success {
+                        case true:
+                            // TODO: call BTSEPADirectDebitClient.tokenize
+                            return
+                        case false:
+                            completion(nil, error)
+                            return
+                        }
+                    }
+                }
```

Comment:
We should probably have a final `else` here that passes a generic error in the completion, even if `error` is nil.

Reply:
Same comment for here, could use guards for both the `error != nil` and the `result` check. 
https://github.com/braintree/braintree_ios/pull/799/files#r833424928